### ISSUE TITLE
Energy randomly stops "piping"

### ISSUE THREAD

**Issue by nonsens28** [mekanism/Mekanism#6356]
#### Issue description: 
Energy cables randomly stop working sometimes , having to replace the malfunctioning one in order to repair the energy chain , and sometimes after replacing it stops 1-2 cables pieces further
![Untitled](https://user-images.githubusercontent.com/68514880/87882956-894a7d00-ca0c-11ea-9031-f1dfae361140.png)

#### Steps to reproduce:
1.
2.
3.

#### Version (make sure you are on the latest version before reporting):
**Forge:** 32.0.67
**Mekanism:** 10.0.2

**Other relevant version:** 

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)

[[gist](https://gist.github.com/) / [pastebin](https://pastebin.com/) / etc link here. Please make sure that it isn't set to expire.]

**Comment by pupnewfster** [comment]
I believe what is happening is that for some reason the networks are not "joining" together, but I am not quite sure why or how as I am unable to reproduce this. I would be interested in seeing the latest log the next time this happens and also what the network reader outputs about the network if used on both the working section of cabling, and the "broken" section of cabling. Also how often does this end up happening or was it only a odd one off occurrence that you haven't had happen again since?

**Comment by nonsens28** [comment]
Ok so i just joined our server , havent played on it after ive put up the
issue ( not because of it tho ) , and now the cables are working properly
because my buddy said he replaced them all ( this did not happen to me ,
only at my buddys side , he also had the chunk problem which u have fixed
on 10.0.2 , like wind mills not working ) . I remember that the network
reader said that the working cables , the green ones had energy , while the
next one , the broken empty one , did not have any energy in it . Now he
told me that it happened quite often when he leaves the chunk and comes
back ( usually teleports to me and goes back to his base ).
Now we are going to play more today so if anything happens i think ill add
on discord and tell you , maybe even share screen you if u want ( im
nonsens28#9441 on discord ) .
And about the latest log , im not the one running the server , its another
buddy , but i could get my hands on it if needed , tho i dont think its
relevant now as the cables are working .
Also i really appreciate your work , and good luck with your school :)

**Comment by archer9234** [comment]
I've ran into this problem on my server. Caused by a different issue. Refined Storage bugged out when I connected an external storage to a Immersive Engineering tank. It broke the entire RS system. The grid wouldn't allow me to access the storage. Or pull things out. The grid would stay on, when it was fully cut off from power and the network. After Disconnecting it from the tank. It got "fixed". But than the power problem with mekanism started. It first happened to the Controller block losing power. The pipe lost power at the connection point, like here. Than followed by all Meka machine blocks losing power. 

I did attempt removing the whole cabling in my base. It didn't fix it permanently, when the server is restarted. Also, the cabling storage amount will keep going up infinity, as you disconnect the problematic section. So say you have a purple cable. The internal buffer is 400MFE. Removing and adding it back on will increase it by 40 units. And you can do that till it has thousands of MFE units. And it will just drain. Even when nothing is using it. The piping problem can also spreading to remaining pipe blocks and other devices that use power, on other mods. Deleting even the machines doesn't help. Either the entire map, or chunk is permanently corrupted.

**Comment by Remlej9** [comment]
This also happens for me on my friends server, and it always happens when we unload the chunks and then load them again. It's a pain to go through the entire cable and reconnecting all the cables. And when I reconnect them at one place it usually breaks at another place. https://imgur.com/a/AmmZApx

**Comment by pupnewfster** [comment]
Ok a few things:
1. What version of Mekanism are you on
2. Just to confirm all these cases are on a server
3. If you could provide a list of mods that would be useful given I am not currently sure (as I haven't been able to reproduce it) if this is mekanism, so getting a list of mods from everyone it is happening to may help narrow down what is actually causing this.

**Comment by archer9234** [comment]
Here's my server mod folder as I left it since the bug occurred: https://imgur.com/CEjg7oW

**Comment by Remlej9** [comment]
I'm on a server running mekanism 10.0.6.429, these are my mods: https://imgur.com/a/U4AKxXh

**Comment by Eztex** [comment]
I have had this issue recurring for several weeks now.  I am not running a server.  Started with the Refined Storage controller losing power but most recently I have noticed several Mekanism machine were losing power.  Advanced Smelter,  Rotary Cond.   etc.  I seems that it will happen either when I quit to title or if I AFK and Minecraft goes to the Game Menu.  My mod list is rather long so I'll see if I can generate an easy list.   Forge 108, 1.16.1

**Comment by calclover2514** [comment]
I am playing Valhelsia 3 and I can confirm that this issue is still happening as of Mekanism v10.0.9.432 with Forge v32.0.108:
![CableOn](https://user-images.githubusercontent.com/43104632/94817455-06aa4a80-03cb-11eb-9261-80a1cfd63ac4.png)
![CableOff](https://user-images.githubusercontent.com/43104632/94817462-08740e00-03cb-11eb-80ce-7a1e74220d1b.png)
This is on a cable crossing ~12 chunks to power a Digital Miner.

**Comment by pupnewfster** [comment]
I would be interested in knowing if anyone is still running into this with Mekanism 10.0.18 as various things have been fixed including some issues related to capabilities/chunk unloading/reloading.

**Comment by IdahoSixString** [comment]
I am playing Valhelsia 3 and I can confirm that this issue is still happening as of Mekanism v10.0.17.444.

**Comment by pupnewfster** [comment]
> I am playing Valhelsia 3 and I can confirm that this issue is still happening as of Mekanism v10.0.17.444.
> 
> If I want things to reconnect I have to break cable and replace it

I said 10.0.18 not 10.0.17, whether it is broken on 10.0.17 or not is *irrelevant* to my question.

**Comment by Viktordrr4** [comment]
Playing Valhelsia 3 with 10.0.18.445 and have this issue.

**Comment by IdahoSixString** [comment]
There I updated to 10.0.18.445 and I still see the issue.

**Comment by codydbgt** [comment]
i have the same issue and they do stop on chunk borders still

**Comment by jwiley17** [comment]
This issue seems to also be apparent on version 10.0.18.445 running on forge 36.0.1

It seems to not only happen with energy cables but any piping from mekanism.

**Comment by IdahoSixString** [comment]
@jwiley17 Agreed it happens with all pipe types. Could you do me a favor and test if putting a digital miner with an anchor tag along these pipes that drop out and see if that fixes the issue for you? 

It fixed the issue for me. Just curious to see if this is just a one off event or there is something to it.

**Comment by thiakil** [comment]
Try without Performant and/or turn things like `Load balancing for AI/Entities, Tileentities and Events` as this will break your pipe networks.

**Comment by MizozPL** [comment]
I have a similar issue. The cable is connected to the generators and the Network Reader clearly shows it has some power stored, but the machine next to it is not getting power.
![image](https://user-images.githubusercontent.com/13796896/110192046-1d4a4700-7e2c-11eb-8b88-4ac4075d74de.png)
I can fix it by placing a single cable block next to another one and breaking it. Then the machine is getting power again.
Notice that the "Acceptors" value changes.
![image](https://user-images.githubusercontent.com/13796896/110192217-f3ddeb00-7e2c-11eb-9065-e87f777ed66d.png)

The issue reappears every time I port to nether and return to my base (it's not chunk-loaded) so I can easily reproduce it on my world save. But I was not able to reproduce in another save.

Mekanism: 10.0.21.448
Forge: 36.0.40

**Comment by pupnewfster** [comment]
out of curiosity if you sneak right click the machine with the network reader does it show that the cable is connected to it? Also are any parts of that cable's network chunk loaded?

**Comment by MizozPL** [comment]
When the glitch happens shift-right clicking the machine with the Network Reader doesn't output anything to the chat. After I place and break a cable to un-glitch the network, it then shows:
![image](https://user-images.githubusercontent.com/13796896/110204211-38d84080-7e72-11eb-94f2-a6193e488b15.png)

I have also removed all the other mods and config folder for Mekanism (to use the default configs). The issue still happens.
Nothing on the entire map is chunk-loaded.

**Comment by PaulEndri** [comment]
We ran into this... Nuking performant out of the server fixed this (and other) issues

**Comment by DBotThePony** [comment]
Can't confirm pipes "disconnecting" from each other, but can confirm that energy cables sometimes refuse to pump energy into machines. Interestingly though, machine and cable are *not* on chunk border.

Can this happen due to how TileEntity getting deserialized on chunk load (their order)?

**Comment by AmirAmnoun** [comment]
Same issue happening on my server
 
Happens when I go far away from an area. When I come back, power isn't transferring through any of the cables.

Most of the type it's enough to replace a single cable in the network, but rarely I have to replace all the cables in the network. 

While I was replacing them it appeared that whenever I replaced a cable, energy would be going through all cables until the last one I replaced (they are green) that cable does however appear to be connected to the next cable (the yet to be replaced one) although energy isn't being transferred between them.


Mekanism Version: 10.0.21.448
MC Version: 1.16.5
Forge Build: 36.1.0

Here's my mod list: https://imgur.com/W4kEw7N

Hope this helps you fix this bug :D

**Comment by JuniorBR790** [comment]
Happening to me.
Valhelsia+3-3.3.2a
Mekanism-1.16.5-10.0.21.448

I tried to remove Performant, no change.
This problem is killing the game experience for me, everything I have that needs energy stops working all the time..
I'm gonna try to reproduce on a new world and give more information here.

**Comment by ivanpu** [comment]
Have this happening to me lately randomly.
I'm using uptodate Valhelsia 3 modpack, in survival single player world.
Probably started to happen around modpack update to V3.3.3 (I've started with 3.3.2), or after I've moved some machines around (digital miner to a neighboring area, factories inside the base, and removed heat generators).
Additional behavior, is that I can change the cable from plain to pushing, and the machine continues to accept the power for a while longer.
Will world folder help you investigate it?

**Comment by Caldoric** [comment]
I'm having a similar issue, but with item, gas, and fluid pipes instead, where the endpoints will randomly reconfigure themselves to either push, pull, or neutral. Also, random breaks in the cable path sometimes just *happen*.

**Comment by Caldoric** [comment]
Perhaps it's time the pipes and cables etc got recoded from the ground up; they've been around a while.

**Comment by IdahoSixString** [comment]
This is such a great mod but this ongoing issue really really hurts this mod.

**Comment by ivanpu** [comment]
Some kind of workaround that I've found:
Splitting the single network into small islands interconnected by entangloporters seems to avoid this issue.
I've also made sure that each such island is confined inside single chunk, but not sure if it's required.

**Comment by Caldoric** [comment]
Tbh, I've had this happen to me on isolated lines, not just part of a network... I once used a set of item transporters to drop stuff from a vanilla funnel to a Create conveyor belt, which worked well enough until the bloody thing reconfigured itself at the "put on the belt" end. Had to re-place half the pipes to get it to work properly. I also think that one of the water pipes we were using on our reactor went fritzy and made it blow up, despite literally everything being placed perfectly a few days back. Unless it's one of those Goblin Traders from one of the other mods in the pack tampering with things... Idk if they can actually do that or not, or if I'm just paranoid, but it's not stated on their resource page(s).

**Comment by IdahoSixString** [comment]
Yeah I had the biggest reactor size go nuclear because the water line stopped working.... I mean it was ehh fun having to move just a lot of work walking in and out of radiation to collect stuff.

**Comment by Caldoric** [comment]
There are commands to remove that...

**Comment by Caldoric** [comment]
Also, did you encase the reactor in obsidian or something similarly blast resistant?

**Comment by Seniorendi** [comment]
Here are... a lot of messages

**Comment by MizozPL** [comment]
> I have a similar issue. The cable is connected to the generators and the Network Reader clearly shows it has some energy stored, but the machine next to it is not getting powered.
> ![image](https://user-images.githubusercontent.com/13796896/110192046-1d4a4700-7e2c-11eb-8b88-4ac4075d74de.png)
> I can fix it by placing a single cable block next to another one and breaking it. Then the machine is getting powered again.
> Notice that the "Acceptors" value changes.
> ![image](https://user-images.githubusercontent.com/13796896/110192217-f3ddeb00-7e2c-11eb-9065-e87f777ed66d.png)
> 
> The issue reappears every time I port to nether and return to my base (it's not chunk-loaded) so I can easily reproduce it on my world save. But I was not able to reproduce in another save.
> 
> Mekanism: 10.0.21.448
> Forge: 36.0.40
> 
> I'm not using Optifine/Performant etc.
> @pupnewfster I can PM you client+server should you need them.

So I've loaded my old world to perhaps help with tracking down the issue. The procedure I use to recreate the issue is the same as mentioned in my previous posts (base->go to nether->let the chunks unload->port back to base through portal).

This time I've found /mek watch command and used it on one of the chunks that had issues with cables failing.
So here it goes:

At first everything is working, acceptor count is 24, as in the image I posted before.
Going into nether - chunk is getting unloaded
`[Server thread/INFO] [Mekanism/]: Dealing with 1 invalid Transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 125 invalid Transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 25 orphan Transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 25 invalid Transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 4 orphan Transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 3 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters`



Getting back to base - chunk is getting loaded
`[Server thread/INFO] [Mekanism/]: Dealing with 133 orphan Transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 116 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 6 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 2 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 2 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 2 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 1 transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 18 orphan Transmitters
[Server thread/INFO] [Mekanism/]: Merging 3 networks with 11 new transmitters
[Server thread/INFO] [Mekanism/]: Adding 3 transmitters to single found network
[Server thread/INFO] [Mekanism/]: Merging 2 networks with 4 new transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 4 invalid Transmitters`
At this point I'm in base, acceptor count is 2, like in the image I posted earlier. Machine is not getting power and is not displaying energy (in the connected sides) when probed with network reader (but still displays gasses as it should).



Placing basic universal cable next to the others.
`[Server thread/INFO] [Mekanism/]: Dealing with 1 orphan Transmitters
[Server thread/INFO] [Mekanism/]: Adding 1 transmitters to single found network`
At this point I'm still not getting power into the machine, neither energy displays as connected side.



Breaking that one additional cable.
`[Server thread/INFO] [Mekanism/]: Dealing with 1 invalid Transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 133 orphan Transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 133 transmitters`
At this point the machine is getting power and right-clicking with network reader at the cable gives the correct acceptor count(24).



I can trigger this bug on demand by going into the nether and coming back to base so feel free to give me any ideas that could help debug it further. The mekanism version is the same as mentioned in my first post.

**Comment by MizozPL** [comment]
What seems strange to me is this part of the log that gets printed after breaking a single cable (my network has over 100 of them).
[Server thread/INFO] [Mekanism/]: Dealing with 1 invalid Transmitters
[Server thread/INFO] [Mekanism/]: Dealing with 133 orphan Transmitters
[Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 133 transmitters

So it looks like it fails to create a network on the chunk load and does so ONLY AFTER I've broken a cable (or am I wrong?).

**Comment by IdahoSixString** [comment]
Nope you have pretty well nailed it on the head and with the logs should help the developer narrow it down further. 

The work around in the mean time is to make sure that chunk is has a machine with the anchor tag in it like the digital miner. That will prevent this problem from occurring.

**Comment by MizozPL** [comment]
> What seems strange to me is this part of the log that gets printed after breaking a single cable (my network has over 100 of them).
> [Server thread/INFO] [Mekanism/]: Dealing with 1 invalid Transmitters
> [Server thread/INFO] [Mekanism/]: Dealing with 133 orphan Transmitters
> [Server thread/INFO] [Mekanism/]: No networks found. Creating new network for 133 transmitters
> 
> So it looks like it fails to create a network on the chunk load and does so ONLY AFTER I've broken a cable (or am I wrong?).

Hmm I'm not so sure about the failing-to-create-network-thing though... Because probing with the network reader shows the correct transmitters count, but the acceptor count is 2 instead of 24.

So my guess is that somehow it fails to load the acceptors on chunk load (bad caching?) **but breaking the universal cable causes to invalidate entire DynamicNetwork and creation of new one.**

**Comment by MizozPL** [comment]
I have managed to make the issue REPRODUCIBLE. Here's my setup

Clean MultiMC Instance
Forge: 36.2.4
Mekanism: 1.16.5-10.0.21.448
Java: jdk-16.0.2+7 from adoptopenjdk.net (hotspot)
Java args: -Xmx1G -Xms1G
Every other config/setting is on default.

Steps to reproduce:
1. Boot Minecraft
2. Create a new world (Creative, Type: SUPERFLAT, Preset String [110*minecraft:bedrock])
3. Teleport to 849.5 111.0 2110.5 (/tp 849 111 2110)
4. Follow exactly this block placement:
  - 2 crushers at: 849 110 2110 and 851 110 2110
  - 7 basic universal cables at: (849 111 2110), (849 111 2111), (849 111 2112), (850 111 2112), (851 111 2112), (851 111 2111), (851 111 2110)
5. Check if the setup resembles the one from the picture and probe the cable with Network Reader. It should say 7 transmitters and 2 acceptors.
6. Build a nether portal with corners at: (849 110 2105) and (852 114 2105)
7. Go into nether
8. Fly for a while to let the chunks unload (That's why we're using Xmx1G and Xms1G)
9. Go back through the portal
10. Probe any of the cables with the Network Reader if it shows 0 acceptors then you have reproduced the bug. If acceptors are still present then repeat from step 7.

Observation: Attaching a new cable to this broken network doesn't seem to update it but breaking one does and fixes the issue.
I have managed to reproduce this issue on Java 8, 15, and 16 so I guess it's not version dependant.

Final Setup Pic:
![FinalSetup](https://user-images.githubusercontent.com/13796896/132956966-8e1d7040-4587-4bbb-bf3c-749df230bcde.png)

**Comment by MizozPL** [comment]
After probing the right cable(attached to one of crushers) in the network's broken state the debugger shows that the cable is having correct cachedAcceptors count but the network shows 0 acceptors in total. So the TileEntity seems to correctly update it's acceptor's count but theNetwork doesn't seem to update it accordingly.

Here's a pic:
![image](https://user-images.githubusercontent.com/13796896/132969683-cb5b983a-0b8b-45ef-a0f7-e236fb954429.png)

Will have to look at the network update code next...

**Comment by MizozPL** [comment]
So here's my solution to this bug

In Transmitter.java replace this:
`@Nonnull
    public LazyOptional<ACCEPTOR> getAcceptor(Direction side) {
        return acceptorCache.getCachedAcceptor(side);
    }`

With this:
`@Nonnull
    public LazyOptional<ACCEPTOR> getAcceptor(Direction side) {
        transmitterTile.tick();
        return acceptorCache.getCachedAcceptor(side);
    }`

I believe that calling single tick on chunk-loading won't affect performance that much. But we can also consider exposing forceUpdate flag and acting accordingly in aforementioned places.

**Comment by MizozPL** [comment]
So the main problem for now is that commiting the DynamicNetwork can happen before acceptorCache is updated in Transmitters. But manually calling refreshConnection() which sets **forceUpdate flag to false causes issues with machines not being able to receive power. I don't know yet why?**

**Comment by MizozPL** [comment]
So despite calling Transmitter.refreshConnections() before DynamicNetwork.commit() happens we're still not getting energy transfer because of the isValid parameter being set to false... At least all acceptors now show in network but some of them (the ones that were not working are flagged as not valid and cannot transfer energy anyways...

**Comment by pupnewfster** [comment]
Thank you for looking into this so extensively @MizozPL. Sometime this week I am going to try and finish rewriting a system I was in the middle of working on when I took a bit of a break from modding, and then afterwards I will look into this further. I don't think your solution is quite right but your analysis of the issue and how to reproduce it looks promising. I am not quite sure yet if it is something I will try and patch, as the network formation code is quite a mess currently and I think it may be more worthwhile for me to try and rewrite it in such a way that would better handle the weird edge case ordering of how things are being ran.

**Comment by MizozPL** [comment]
> Thank you for looking into this so extensively @MizozPL. Sometime this week I am going to try and finish rewriting a system I was in the middle of working on when I took a bit of a break from modding, and then afterwards I will look into this further. I don't think your solution is quite right but your analysis of the issue and how to reproduce it looks promising. I am not quite sure yet if it is something I will try and patch, as the network formation code is quite a mess currently and I think it may be more worthwhile for me to try and rewrite it in such a way that would better handle the weird edge case ordering of how things are being ran.

Sure, the solution isn't right as per my last post - it only makes acceptors always register in the network but for some reason the isValid flag of LazyOptional<ACCEPTOR> is set to false. So yes they show up in network now, but energy still does not flow. Now I'm trying to figure out what can cause this flag to be false.

So it isn't as simple as forcing a refresh on Transmitter before accessing it's cache, because acceptor that was queried that way gets invalidated...

**Comment by MizozPL** [comment]
Ok so a bit more digging and I found that the following happens (using my test setup):
- There are 4 transmitters without a network
- 2 networks are being created (with 2 transmitters each)
- DynamicNetwork.acceptorChanged() is being called for each (since wwe have 2 crushers) and so those 2 networks are added to networksToChange in TransmitterNetworkRegistry
- onTick is called and before networksToChange is allowed to be processed (adding missing acceptors) **getNetworkFromOrphan() is merging those two into a new one without considerring those two needed an update**

I think we should call `commitChanges();` before `network = finder.createNetworkByMerging();`
And so calling it fixed the issue in my test setup! But I didn't test it for too long...

**Comment by MizozPL** [comment]
#7308 Possible fix

**Comment by pupnewfster** [comment]
10.0.22 has been [released](https://www.curseforge.com/minecraft/mc-mods/mekanism/files/3497015) and should fix this. I am going to close this for now, but if anyone is still running into this issue with 10.0.22, please comment and we will reopen this.

**Comment by pupnewfster** [comment]
@Forzii (or anyone who reliably is reproducing this with 10.0.22) could you try replacing the Mekanism-1.16.5-10.0.22.449 version on your server with [this build](https://drive.google.com/file/d/1wmmNB9iViWmGHC21LWlKrB0p6WlbDD-s/view?usp=sharing) and let me know if it fixes the problem? If it does I will commit the changes I made, and release a 10.0.23 build.

**Comment by jexom** [comment]
Doesn't seem like it. Switched out the jar, replaced all the cables, went to the end for some resources and when i got back all the cables are on the separate networks

**Comment by pupnewfster** [comment]
@jexom would you mind trying [this build](https://drive.google.com/file/d/1a7HI0rukvh0F95gZIlSN1nW_UcZMn4b2/view?usp=sharing) and sending me the server's latest log after it breaks again? That build *won't* fix the issue but I added some extra logging that hopefully will confirm if I am on the right track with my thinking of what the issue may be.

**Comment by Seniorendi** [comment]
Fixing this one-year-old issue will be amazing <:

**Comment by pupnewfster** [comment]
Hmm, ya you definitely are right that the logging messages I added didn't show up, which means that those potential edge cases I was wondering if are not being handled properly aren't the cause of the issue, so I will keep looking to see if I can come up with any other ideas what areas seem like they may have some issues.

**Comment by pupnewfster** [comment]
@jexom There is a chance [this build](https://drive.google.com/file/d/1jIyF3rfl2cci_LiGkGkD45NlkfgfBqhp/view?usp=sharing) will fix the issue (though at this point who knows), at the very least it hopefully will print some messages to the log that might help me track the issue down better/know what it was, as I added a fair bit more logging to it.

**Comment by jexom** [comment]
This may have fixed it. Still didn't see anything different within the logs but couldn't reproduce the issue. Here are the logs just in case
[latest.log](https://github.com/mekanism/Mekanism/files/7469169/latest.log)
[debug.log](https://github.com/mekanism/Mekanism/files/7469170/debug.log)

**Comment by pupnewfster** [comment]
Hmm, I will do a bit more cleanup then and probably try to release an actual build either later today or tomorrow, going off the assumption that it is fixed now if you can't reproduce it anymore. I am sort of surprised that none of the messages I added ended up getting logged, but I did do a bit of general cleanup to network registration, so it is possible one of the things I fixed while cleaning it up ended up fixing this issue as a side effect.

**Comment by pupnewfster** [comment]
Lets try this again... 10.0.23 is now released, so I am going to close this, if it turns out it isn't actually fixed comment and we will open this yet again.

**Comment by pupnewfster** [comment]
https://discord.com/channels/554379889122213888/560583846509281280/912041633334296667 Seems like this may not be entirely fixed after all...

**Comment by sippeeey** [comment]
It seems to be the same problem in version 10.0.24.
It is randomly, sometimes the multi block structures are connected and any other are disconnected. Is also happens with the "induction casing battery" or "Thermal Evaporation Tower"

![2021-12-12 12_25_32-Minecraft_ 1 16 5 - Multiplayer (3rd-party Server)](https://user-images.githubusercontent.com/38763652/145710365-eba3295c-79f9-4726-af1b-811d1dd87a87.png)



And its fixed randomly after Server restart. Before i restarted, the i had fix the left tower by replace on block from the multi block, but the right tower was broken. after restart also the right tower was fixed, although i dont replace a block from the multi block structure. 

! To Replace a cable on the port doesnt fix anythere.

![2021-12-12 12_32_58-Minecraft_ 1 16 5 - Multiplayer (3rd-party Server)](https://user-images.githubusercontent.com/38763652/145710657-f4a65021-2491-433a-9f60-7d684dbd803d.png)

Server java version: 
_openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment (build 16.0.1+9-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 16.0.1+9-Ubuntu-120.04, mixed mode, sharing)_

Client java verison:
_Java 1.8.0_51 64bit_

Minecraft Server version:
_mohist-1.16.5-778-server.jar_

Modlist:
_ae2wtlib-0.3.2-1.16.5.jar
alexsmobs-1.12.1.jar
appleskin-forge-mc1.16.x-2.2.0.jar
appliedenergistics2-8.4.4.jar
Aquaculture-1.16.5-2.1.21.jar
BetterThirdPerson-Forge-1.16.4-1.5.1.jar
BiomesOPlenty-1.16.5-13.1.0.477-universal.jar
blockcarpentry-1.16-0.4.0.jar
Bookshelf-Forge-1.16.5-10.3.29.jar
carryon-1.16.5-1.15.5.15.jar
cfm-7.0.0pre22-1.16.3.jar
cgm-1.0.1-1.16.3.jar
Chisel-MC1.16.5-2.0.1-alpha.4.jar
chiselsandbits-1.0.43.jar
ChunkAnimator-1.16.5-1.2.4.jar
chunkloaders-1.1.7-mc1.16.5.jar
citadel-1.8.1-1.16.5.jar
CodeChickenLib-1.16.5-4.0.4.435-universal.jar
CraftTweaker-1.16.5-7.1.2.468.jar
create-mc1.16.5_v0.3.2g.jar
CreativeCore_v2.2.1_mc1.16.5.jar
CTM-MC1.16.1-1.1.2.6.jar
Cucumber-1.16.5-4.1.12.jar
Cyclic-1.16.5-1.5.11.jar
DynamicTrees-1.16.5-0.10.0-Beta25.jar
DynamicTreesBOP-1.16.5-2.0.6.jar
DynamicTreesPlus-1.16.5-0.1.0-Beta10.jar
elevatorid-1.16.5-1.7.13.jar
EnderStorage-1.16.5-2.8.0.168-universal.jar
fishingoverhaul-1.16.5-1.0.2.jar
FluxNetworks-1.16.5-6.1.7.12.jar
flywheel-1.16-0.2.5.jar
gravestone-1.16.5-1.0.6.jar
Hwyla-forge-1.10.11-B78_1.16.2.jar
invtweaks-1.16.4-1.0.1.jar
ItemPhysic_v1.4.18_mc1.16.5.jar
jei-1.16.5-7.7.1.137.jar
JEITweaker-1.16.5-1.0.1.35.jar
JustEnoughResources-1.16.5-0.12.1.128.jar
kotlinforforge-1.16.0-obf.jar
lemonlib-1.3.6.jar
Mantle-1.16.5-1.6.127.jar
Mekanism-1.16.5-10.0.24.453.jar
MekanismAdditions-1.16.5-10.0.24.453.jar
MekanismGenerators-1.16.5-10.0.24.453.jar
morecfm-1.3.1-1.16.3.jar
Morpheus-1.16.5-4.2.70.jar
MouseTweaks-2.14-mc1.16.2.jar
MysticalAgriculture-1.16.5-4.2.5.jar
obfuscate-0.6.2-1.16.3.jar
OreExcavation-1.8.157.jar
simplegens-1.16.5-3.0.9.2.jar
Space-Bosstools-1.16.5-5.4.jar
StorageDrawers-1.16.3-8.3.0.jar
supermartijn642configlib-1.0.9-mc1.16.jar
TConstruct-1.16.5-3.2.1.296.jar
TravelersBackpack-1.16.5-5.4.5.jar
valkyrielib-1.16.5-3.0.9.5.jar
vehicle-mod-0.45.2-1.16.3.jar
WailaHarvestability-mc1.16.x-forge-1.1.15.jar
WAWLA-1.16.5-8.0.1.jar
Xaeros_Minimap_21.22.2_Forge_1.16.5.jar
XaerosWorldMap_1.18.3_Forge_1.16.5.jar
xlfoodmod-1.16.4-1.2.0.jar_

**Comment by pupnewfster** [comment]
My guess is this isn't the case but I forgot to actually confirm it with the person who reported still having this in 10.0.24 on discord, but @sippeeey are you able to reproduce it if you test without Mohist?

**Comment by Icron387** [comment]
Problem still exists in Mekanism 10.2.2

**Comment by Seniorendi** [comment]
This bug tastes good

**Comment by pupnewfster** [comment]
I believe this is fixed in 10.3.2 (for MC versions 1.19.1 and 1.19.2), if anyone can reproduce it on that version or newer please let me know and I will reopen the issue. Unfortunately due to the changes required for the fix it is not practical/feasible to backport the fix to unsupported MC versions (1.16 and 1.18).

**Comment by MenJoe** [comment]
On MC 1.20.1, mekanism 10.4.5 and  forge 47.2.16 it happens again. Im playing on ATM 9 version 0.2.33 and using FTB chunks to load base sometimes when i log back to server cables stop transmiting power, breaking 1 cable and placing it back sometimes fix it or propagate it to next cable. One fix to that i found is unload everyting, move away and load chunks again.

**Comment by Caldoric** [comment]
Seriously, you'd think they would've fixed this by now, seeing as it's an issue that's been around for at least a couple years.

**Comment by SirEndii** [comment]
Tbf I never had this issue in my worlds
But I also don't have large cable networks, I use quantum entangloporters a lot

**Comment by ohm-ena** [comment]
Still happens, currently on **ATM9 version 0.3.5, MC 1.20.1, Forge 47.3.11**, super small setup: one solar panel hooked up to an ultimate energy cube, and from there one longish cable connecting to Refined Storage controller + 8 Mekanism machines and a chargepad. Had to replace all the cables since replacing the faulty one propagated it to the next one.

**Comment by Caldoric** [comment]
Argh, how is this bug still a thing, several years later? I'm beginning to wonder if it's not directly something in the cables themselves, but something elsewhere that points _at_ the cables themselves in the code instead...?

**Comment by space-agent** [comment]
still happens 1.20.1 10.4.11.67

**Comment by o0omega** [comment]
still an issue

**Comment by o0omega** [comment]
![Image](https://github.com/user-attachments/assets/4b7f2111-f57b-44fa-8034-58316f7d0d2e)

**Comment by thiakil** [comment]
Please create a new issue for new occurrences, after removing any mods that add on or tweak Mekanism made by other teams.
